src/battle_ai_util.c

```diff
@@ -3735,11 +3735,14 @@ void IncreaseBurnScore(u32 battlerAtk, u32 battlerDef, u32 move, s32 *score)
 
     if (AI_CanBurn(battlerAtk, battlerDef, AI_DATA->abilities[battlerDef], BATTLE_PARTNER(battlerAtk), move, AI_DATA->partnerMove))
     {
-        ADJUST_SCORE_PTR(WEAK_EFFECT); // burning is good
-        if (HasMoveWithCategory(battlerDef, DAMAGE_CATEGORY_PHYSICAL))
+        if (HasMoveWithCategory(battlerDef, DAMAGE_CATEGORY_PHYSICAL)
+            || (!(AI_THINKING_STRUCT->aiFlags[battlerAtk] & AI_FLAG_OMNISCIENT) // Not Omniscient but expects physical attacker
+                && gSpeciesInfo[gBattleMons[battlerDef].species].baseAttack >= gSpeciesInfo[gBattleMons[battlerDef].species].baseSpAttack + 10))
         {
-            if (CanTargetFaintAi(battlerDef, battlerAtk))
+            if (CanTargetFaintAi(battlerDef, battlerAtk) && gMovesInfo[GetBestDmgMoveFromBattler(battlerDef, battlerAtk)].category == DAMAGE_CATEGORY_PHYSICAL)
```

Comment:
is the `CanTargetFaintAi` check needed? I would say it is always useful to burn if the best damage move is physical.

Reply:
The initial code was intending to uniquely give a score increase if the player's mon was threatening a KO pre-burn, whereas removing that check would always give the biggest score increase if the player's best move was physical, even if it only did 1 damage as opposed to 0 damage from its other moves.

Reply:
Yes, but is the `CanTargetFaintAi` check useful in general? Should the condition best damage move is physical not be enough for a good increase?

Reply:
OH I understand what you mean. Yeah I am on board with that for sure, will change :)